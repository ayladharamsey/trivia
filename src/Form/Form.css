form {
    /* height: 60%; */
    height: 60vw;
    width: 30%;
    border: 3px solid #66FCF1;
    border-radius: 5px;
    background-color: #0B0C10;
    margin-left: 35%;
    padding: 20px;
  }

.submit {
    display: block;
    margin-left:37%;
    margin-top: 30px;
}

img {
    height: 25%;
}

input:focus,
select:focus {
    outline: none;
    box-shadow: 0px 0px 15px red;
}

.select select:active {
    color: #fff;
    background-color: #0074d9;
}

.select:after {
    position: absolute;
    top: 50%;
    right: 1.25rem;
    display: inline-block;
    content: "";
    width: 0;
    height: 0;
    margin-top: -.15rem;
    pointer-events: none;
    border-top: .35rem solid;
    border-right: .35rem solid transparent;
    border-bottom: .35rem solid transparent;
    border-left: .35rem solid transparent;
}

.select select:focus:-moz-focusring {
  color: transparent;
  text-shadow: 0 0 0 #000;
}

.select {
  position: relative;
  display: inline-block;
  width: 52%;
  color: #555;
}
.select select {
  display: inline-block;
  width: 100%;
  margin: 0;
  padding: .5rem 2.25rem .5rem 1rem;
  line-height: 1.5;
  color: #555;
  background-color: #eee;
  border: 0;
  border: 1px solid #D0D0D0;
  border-radius: .25rem;
  cursor: pointer;
  outline: 0;
  -webkit-appearance: none;
     -moz-appearance: none;
          appearance: none;
}
 