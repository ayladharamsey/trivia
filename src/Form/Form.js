import React, { Component } from 'react';
import './Form.css';


class Form extends Component {
    constructor() {
        super();
        this.state = {
            name: '',
            quote: '',
            rating: ''
        } 
    }

    submitInfo = (event) => {
        this.setState({[event.target.name]: [event.target.value]})
    }

    handleSubmit = event => {
        this.props.updateState(this.state)
        this.setState({
            name: '',
            quote: '',
            rating: ''
        });

        //then it navigates us to the next page
        
    }

    render() {
        return(
            <section>
                <img alt="logo" src="/images/star_wars_logo_PNG42.png"/>
                <h1>Welcome to SWAPI!</h1>
                <h3>Name </h3>
                <input 
                    type='text' 
                    placeholder="Enter name" 
                    onChange={this.submitInfo} 
                    name="name"
                    value={this.state.name}
                    />
                <h3>Quote </h3>
                <input 
                    type='text' 
                    placeholder="Enter quote" 
                    onChange={this.submitInfo}
                    name="quote"
                    value={this.state.quote} 
                    />
                <h3> Select Jedi Rating </h3>
                <select 
                    name='rating' 
                    size='3' 
                    onChange={this.submitInfo}
                    value={this.state.rating}
                    >
                    <option name='rating'value='apprentice' onClick={this.handleSubmit}>Apprentice</option>
                    <option value='knight' onClick={this.handleSubmit}>Knight</option>
                    <option value='master' onClick={this.handleSubmit}>Master</option>
                </select>
                <button onClick={this.handleSubmit}>Submit</button>
            </section>
        )
    }
}

export default Form;