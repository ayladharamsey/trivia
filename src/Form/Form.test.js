import React from 'react';
import { shallow } from 'enzyme';
import Form from './Form';

describe('Form', () => {
  let wrapper; 

  beforeEach(() => {
    wrapper = shallow(<Form updateAppState = {jest.fn()}/>) 
  });

  it('should match the snapshots', () => {
    expect(wrapper).toMatchSnapshot();
  });


});