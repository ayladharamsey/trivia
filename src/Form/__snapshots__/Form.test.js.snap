// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should match the snapshots 1`] = `
<form>
  <img
    alt="star wars logo"
    src="logo.png"
  />
  <h3>
    Name 
  </h3>
  <input
    name="name"
    onChange={[Function]}
    placeholder="Enter name"
    type="text"
    value=""
  />
  <h3>
    Quote 
  </h3>
  <input
    name="quote"
    onChange={[Function]}
    placeholder="Enter quote"
    type="text"
    value=""
  />
  <h3>
     Select Jedi Rating 
  </h3>
  <div
    className="select"
  >
    <select
      name="rating"
      onChange={[Function]}
      value="Select Rating"
    >
      <option
        disabled={true}
        value="Select Rating"
      >
        Select Rating
      </option>
      <option
        value="apprentice"
      >
        Apprentice
      </option>
      <option
        value="knight"
      >
        Knight
      </option>
      <option
        value="master"
      >
        Master
      </option>
    </select>
  </div>
  <h5 />
  <Link
    to="/movies"
  >
    <button
      className="submit"
      disabled={true}
      onClick={[Function]}
    >
      Submit
    </button>
  </Link>
</form>
`;
